internal/pkg/deploy/deploy.go

```diff
@@ -121,23 +121,44 @@ func (s *Store) ListEnvironmentsDeployedTo(appName string, svcName string) ([]st
 	if err != nil {
 		return nil, fmt.Errorf("list environment for app %s: %w", appName, err)
 	}
-	var envsWithDeployment []string
+	type result struct {
+		name string
+		err  error
+	}
+	deployedEnv := make(chan result, len(envs))
+	defer close(deployedEnv)
 	for _, env := range envs {
-		err := s.newRgClientFromRole(env.ManagerRoleARN, env.Region)
-		if err != nil {
-			return nil, err
-		}
-		resources, err := s.rgClient.GetResourcesByTags(ecsServiceResourceType, map[string]string{
-			AppTagKey:     appName,
-			EnvTagKey:     env.Name,
-			ServiceTagKey: svcName,
-		})
-		if err != nil {
-			return nil, fmt.Errorf("get resources by Copilot tags: %w", err)
+		go func(env *config.Environment) {
+			err := s.newRgClientFromRole(env.ManagerRoleARN, env.Region)
+			if err != nil {
+				deployedEnv <- result{err: err}
+				return
+			}
+			resources, err := s.rgClient.GetResourcesByTags(ecsServiceResourceType, map[string]string{
+				AppTagKey:     appName,
+				EnvTagKey:     env.Name,
+				ServiceTagKey: svcName,
+			})
+			if err != nil {
+				deployedEnv <- result{err: fmt.Errorf("get resources by Copilot tags: %w", err)}
+				return
+			}
+			// If no resources found, the resp length is 0.
+			var res result
+			if len(resources) != 0 {
+				res.name = env.Name
+			}
+			deployedEnv <- res
```

Comment:
This method is getting quite long - just for readability can we separate this inline function into its own function? Then the parent function can mostly worry about doing the boring concurrency bits. 

I'm thinking something in the shape of this - what do you think? (not exactly like this because i didn't think it through too hard - but this shape for sure). 
```golang
func (s *Store) deployedServices(envClient *Client, app, env string) []string, error {
  ...
}

func (s *Store) ListEnvironmentsDeployedTo(appName string, svcName string) ([]string, error) {
  //...
  for _, env := range envs {
    go func() {
      deployedEnv <- s.deployedServices(...)
    } 
  } 
}
```